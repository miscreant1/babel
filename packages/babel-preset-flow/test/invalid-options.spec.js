import { loadOptions } from "@babel/core";
import presetFlow from "..";

function loadPresetWithOptions(options: any) {
  loadOptions({
    filename: "/fake/test.js",
    presets: [[presetFlow, options]],
  });
}
describe("preset-flow", () => {
  it("should throw when top level options are invalid", () => {
    expect(() => {
      loadPresetWithOptions({ All: true });
    }).toThrowErrorMatchingSnapshot();
  });
  it("should throw when boolean options are not provided with boolean values", () => {
    expect(() => {
      loadPresetWithOptions({ all: "true" });
    }).toThrowErrorMatchingSnapshot();
  });
});
