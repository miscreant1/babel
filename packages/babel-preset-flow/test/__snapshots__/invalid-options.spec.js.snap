// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`preset-flow should throw when boolean options are not provided with boolean values 1`] = `"[BABEL] /fake/test.js: @babel/preset-flow: 'all' option must be a boolean. (While processing: \\"base$0\\")"`;

exports[`preset-flow should throw when top level options are invalid 1`] = `
"[BABEL] /fake/test.js: @babel/preset-flow: 'All' is not a valid top-level option.
Maybe you meant to use 'all'? (While processing: \\"base$0\\")"
`;
